### ISSUE TITLE
KeyError

### ISSUE THREAD

**Issue by mingming36** [SY-Xuan/IICS#4]
File "/home/l603a/mrp/II/reid/models/backbones/resnet.py", line 214, in load_param
    self.state_dict()[i].copy_(param_dict[i])
KeyError: 'state_dict'

**Comment by SY-Xuan** [comment]
Could you please provide more information about how you meet this problem.
I think you may use the code in the wrong way.

**Comment by mingming36** [comment]
I just adjust the content in train_market.sh.
export PYTHONPATH=$PYTHONPATH:./
python ./example/iics.py --dataset market1501 --checkpoint "/home/l603a/mrp/II/script/papa/market.pth.tar"

**Comment by SY-Xuan** [comment]
If you want to evaluate, you should add --evaluate. If you want to train, the checkpoint needs to be pretrained resnet.

**Comment by mingming36** [comment]
ok, thanks very much. I will try.

**Comment by stoneyang** [comment]
Hi, @SY-Xuan , could you please specify the pretrained resnet50 you use in your work? Because when loading the weights from the PyTorch Official one, it fails with the following error, which indicates an obvious incorrect configuration: 
```
Traceback (most recent call last):
  File "./example/iics.py", line 394, in <module>
    main(parser.parse_args())
  File "./example/iics.py", line 148, in main
    model.model.load_param(args.checkpoint)
  File "/home/Code/IICS/reid/models/backbones/resnet.py", line 213, in load_param
    self.state_dict()[i].copy_(param_dict[i])
KeyError: 'conv1.0.weight'
```

Thanks in advance! 

Cheers!

**Comment by SY-Xuan** [comment]
Did you modify the original code? I just use the pytorch official one and it works well for me.

**Comment by SY-Xuan** [comment]
I think you use the wrong pre-trained model. This key should be "conv1.weight".
You can download the correct model from https://download.pytorch.org/models/resnet50-19c8e357.pth.

**Comment by stoneyang** [comment]
@SY-Xuan eh..., after posting that, I've examined the keys in the `state_dict` and found the problem is the deprecated pretrained model from PyTorch official! The string is `528c19ca` and this file only support older versions of PyTorch. As a re-training, the result seems correct now: 
```
Mean AP: 74.5%
CMC Scores  market1501
  top-1          89.9%
  top-5          95.3%
  top-10         97.0%
```

Thanks for your timely reply and advice! 

PS: The weights I used in the **successful** experiments: https://download.pytorch.org/models/resnet50-0676ba61.pth
